nodes/src/nodes/tool_filesystem/IInstance.py

```
@@ -0,0 +1,328 @@
+
+
+def _optional_str(args: dict, key: str, *, default: str | None = None) -> str | None:
+    val = args.get(key, default)
+    if val is None:
+        return default
+    if not isinstance(val, str):
+        raise ValueError(f'{key} must be a string')
+    return val
+
+
+def _run_async(coro):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `_run_async`

(ANN202)